### ISSUE TITLE
Windows Support

### ISSUE THREAD

**Issue by 3442853561** [theduke/quickjs-rs#18]
I noticed that this project does not currently support windows. Do we have such a plan?

**Comment by theduke** [comment]
Definitely.
In theory quickjs should already work on Windows. Should be just a matter of updating `libquickjs-sys/build.rs` but I don't have a Windows machine currently , so can't test it myself.

**Comment by 3442853561** [comment]
@theduke I will help you to test it.

**Comment by theduke** [comment]
Building on latest Windows fails on CI with:

```
C:\Users\VSSADM~1\AppData\Local\Temp\ccRuWz3j.s: Assembler messages:
C:\Users\VSSADM~1\AppData\Local\Temp\ccRuWz3j.s:67812: Error: .seh_savexmm offset is negative
C:\Users\VSSADM~1\AppData\Local\Temp\ccRuWz3j.s:67813: Error: .seh_savexmm offset is negative
C:\Users\VSSADM~1\AppData\Local\Temp\ccRuWz3j.s:67818: Error: .seh_savexmm offset is negative
```

**Comment by theduke** [comment]
@3442853561 I'm not sure if there is anything that can be done here, and like I said I can't test further myself.
Did you manage to compile quickjs on your system?

**Comment by 3442853561** [comment]
I failed when I compiled it with MinGW many times. There is a version compiled with TDM-GCC that is available. And I have already tried it. https://github.com/3442853561/quickjs-windows

**Comment by undefinedbuddy** [comment]
Hi!

It seems like [this project](https://github.com/mengmo/QuickJS-Windows-Build) shows how to build QuickJS on Windows. I tested the prebuilt binaries and it seems like everything is working just fine. Perhaps `libquickjs-sys/build.rs` could be updated to assume an `MSYS2` environment with `MINGW64` installed and the appropriate toolchains for 64-bit (`x86_64-toolchain`) and 32-bit (`i686-toolchain`) available. Any ideas?

**Comment by theduke** [comment]
Resolved in #70 .

Windows support is now published in `0.3.3`, with MSYS2 and `x86_64-pc-windows-gnu` target.

It would of course be nice to also support the `msvc`  but I guess that's an upstream issue.

**Comment by Kreijstal** [comment]
```
error: failed to run custom build command for `libquickjs-sys v0.9.0`

Caused by:
  process didn't exit successfully: `C:\Users\kreij\scoop\apps\msys2\2022-01-28\tmp\cargo-installterPrF\release\build\libquickjs-sys-5f3b398ffe6ff8d2\build-script-build` (exit code: 101)
  --- stderr
  Applying patches...
  Applying "js-tobigint64-overflow.patch"...
  thread 'main' panicked at 'Could not apply patches: Error { kind: NotFound, message: "program not found" }', C:\Users\kreij\.cargo\registry\src\github.com-1ecc6299db9ec823\libquickjs-sys-0.9.0\build.rs:133:14
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
error: failed to compile `pandoc-katex v0.1.9`, intermediate artifacts can be found at `C:\Users\kreij\scoop\apps\msys2\2022-01-28\tmp\cargo-installterPrF`
```
failed trying to compile in mingw, hmm?
